apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -21,16 +21,55 @@ import (
 	"github.com/juju/juju/state"
 )
 
+// UserService defines the methods to operate with the database.
+type UserService interface {
+}
+
 // UserManagerAPI implements the user manager interface and is the concrete
 // implementation of the api end point.
 type UserManagerAPI struct {
-	state      *state.State
-	pool       *state.StatePool
-	authorizer facade.Authorizer
-	check      *common.BlockChecker
-	apiUser    names.UserTag
-	isAdmin    bool
-	logger     loggo.Logger
+	state       *state.State
+	userService UserService
+	pool        *state.StatePool
+	authorizer  facade.Authorizer
+	check       *common.BlockChecker
+	apiUser     names.UserTag
+	isAdmin     bool
+	logger      loggo.Logger
+}
+
+// NewAPI creates a new API endpoint for calling user manager functions.
+func NewAPI(
+	ctx facade.Context,
+	userService UserService,
+) (*UserManagerAPI, error) {
```

Comment:
We should pass in all the dependencies here. I know it's painful, but is the correct thing to do.

For example; `authorizer`, `state`, `logger` and the `statePool`.

Reply:
done